src/test/scala/chiselTests/Direction.scala

```diff
@@ -379,4 +379,46 @@ class DirectionSpec extends ChiselPropSpec with Matchers with Utils {
     assert(emitted.contains("io.monitor.valid <= io.driver.valid"))
     assert(emitted.contains("io.monitor.ready <= io.driver.ready"))
   }
+  property("Bugfix: marking Vec fields with mixed directionality as Output clears inner directions") {
+    class Decoupled extends Bundle {
+      val bits = UInt(3.W)
+      val valid = Bool()
+      val ready = Flipped(Bool())
+    }
+    class DecoupledAndMonitor extends Bundle {
+      val driver = Output(Vec(1, new Decoupled()))
+    }
+    class MyModule extends RawModule {
+      val io = IO(new DecoupledAndMonitor())
+    }
+
+    val emitted: String = ChiselStage.emitChirrtl(new MyModule)
+
+    assert(
+      emitted.contains(
+        "output io : { driver : { bits : UInt<3>, valid : UInt<1>, ready : UInt<1>}[1]}"
+      )
+    )
+  }
+  property("Bugfix: clearing all flips inside an opaque type") {
+    class Decoupled extends Bundle {
+      val bits = UInt(3.W)
+      val valid = Bool()
+      val ready = Flipped(Bool())
+    }
+    class DecoupledAndMonitor extends Bundle {
+      val driver = new Decoupled()
+    }
+    class MyModule extends RawModule {
+      val w = Wire(Waivable(new Decoupled(), true, true))
```

Comment:
This mentions "opaque type" but I don't see any opaque typing here.

Reply:
Whoops, apparently I wrote my unit test using an opaque type that I didn't move in. I'll fix this.